### PR TITLE
doc: matter: rewrite matter dev environement docs

### PR DESCRIPTION
**Pull request by greg-fer**
Updated Matter dev environment docs for clarity.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/ug_matter_configuring_protocol.rst [hunk 1]
```diff
@@ -53,23 +43,41 @@ This option sets the :kconfig:`CONFIG_OPENTHREAD_MTD_SED` Kconfig option.
...
 =====================
 
 You can enable the Matter shell library using the :kconfig:`CONFIG_CHIP_LIB_SHELL` Kconfig option.
 This option lets you use the Matter shell commands with :ref:`matter_samples`.
 
 See :doc:`matter:nrfconnect_examples_cli` in the Matter documentation for the list of available Matter shell commands.
+
+.. _ug_matter_configuring_requirements:
+
+Required components for Matter network
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_configuring_protocol.rst:75]: Is this section really required? The page is about configuring a Matter application which is unrelated to Matter controllers.

File: doc/nrf/ug_matter_configuring.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
+.. _ug_matter_configuring:
+
+Configuring Matter
+##################
+
+When you program any of the available Matter samples to the supported development kits, you automatically build the Matter stack for the nRF Connect platform.
+The development kit and the application running Matter stack that is programmed on the development kit together form the Matter accessory device.
+
+For the |NCS|, the default use case of a Matter accessory device is within a Matter network that is built on top of a low-power, 802.15.4-compatible Thread network.
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_configuring.rst:9]: I think we could make this stronger as it's not default use case - it's the only use case :) Something like "Currently, NCS only supports Matter stack built on top of a low-power..." would be more accurate.

File: doc/nrf/ug_matter_configuring.rst [hunk 3]
```diff
@@ -0,0 +1,59 @@
+.. _ug_matter_configuring:
+
+Configuring Matter
+##################
+
+When you program any of the available Matter samples to the supported development kits, you automatically build the Matter stack for the nRF Connect platform.
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_configuring.rst:6]:
```suggestion
When you build any of the available Matter samples, you automatically build the Matter stack for the nRF Connect platform.
```

File: doc/nrf/ug_matter_configuring.rst [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+To enable the IPv6 communication with the Matter accessory device over the Thread network, the Matter controller requires the Thread Border Router.
+This is because the Matter controller types described on this page do not have an 802.15.4 Thread interface.
+The Border Router bridges the Thread network with the network interface of the controller, for example Wi-Fi.
+
+Matter development environment setup options depend on how you run the Matter controller.
+You can either run it on the same device as the Thread Border Router or run the Matter controller and the Thread Border Router on separate devices.
+
+Depending on your choice, you can set up one of the following development environments:
+
+* Thread Border Router and mobile Matter Controller
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_configuring.rst:35]:
```suggestion
* Thread Border Router on Raspberry Pi and mobile Matter Controller
```

File: doc/nrf/ug_matter_configuring.rst [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+  .. include:: /ug_matter_configuring_env.rst
+     :start-after: matter_env_ctrl_mobile_start
+     :end-before: matter_env_ctrl_mobile_end
+
+* Thread Border Router and Matter Controller on PC
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_configuring.rst:41]:
```suggestion
* Thread Border Router on Raspberry Pi and Matter Controller on PC
```
It wasn't clear to me if it means that both TBR and the controller run on PC or not.

File: doc/nrf/ug_matter_configuring_env.rst [hunk 6]
```diff
@@ -7,114 +7,182 @@ Configuring Matter development environment
...
+===========================
 
 .. note::
     The Python Matter controller is currently not supported for building on Raspbian OS.
     If you use Raspberry Pi to run the controller and the Thread Border Router, replace Raspbian OS with Ubuntu Server.
+
+To configure and use Thread Border Router and Matter controller on the same device, complete the following steps:
+
+1. Program the Matter accessory device with one of available :ref:`matter_samples`.
+#. Configure the Thread Border Router on a PC or on a Raspberry Pi, depending on what hardware you are using.
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_configuring_env.rst:174]: Recommend using OTBR Docker instead of installing it natively?

**Review by Damian-Nordic — COMMENTED** [review]
A few suggestions from me.

File: doc/nrf/ug_matter_configuring_protocol.rst (hunk 1 above)
Comment (greg-fer) [reply doc/nrf/ug_matter_configuring_protocol.rst:75]: This is meant as "what I do next" section. Maybe we need to name it differently.

Comment (Damian-Nordic) [reply doc/nrf/ug_matter_configuring_protocol.rst:75]: Yeah, but if someone reads about changing the Matter app configuration he's probably already done with the environment setup, so I wonder if this information is useful.

Comment (Damian-Nordic) [reply doc/nrf/ug_matter_configuring_protocol.rst:75]: But it's just my opinion. Feel free to disregard :)

File: doc/nrf/ug_matter_configuring_env.rst [hunk 7]
```diff
@@ -7,114 +7,182 @@ Configuring Matter development environment
...
-To pair and control the Matter accessory device remotely over such a network, you need to set up a Matter controller on a mobile or a PC.
-A Matter controller is a role within the Matter environment.
-This controller interacts with the accessory devices using the following protocols:
+During the last part of the commissioning procedure (the provisioning operation), the Matter controller sends the Thread network credentials to the Matter accessory device.
+As a result, the device can join the Thread network and communicate with other Thread devices in the network.
 
-* Bluetooth® LE during the commissioning process - to securely pass the network credentials and provision the accessory device into the Thread network.
-* Regular IPv6 communication after the accessory device joins the Thread network - to interact with each other by exchanging application messages.
-  For example, to report temperature measurements of a sensor.
+To start the commissioning procedure, the controller must get the commissioning information from the Matter accessory device.
```
Comment (LuDuda) [inline doc/nrf/ug_matter_configuring_env.rst:17]:
```suggestion
To start the commissioning procedure, the controller must get the onboarding information from the Matter accessory device.
```

File: doc/nrf/ug_matter_configuring_env.rst [hunk 8]
```diff
@@ -7,114 +7,182 @@ Configuring Matter development environment
...
 
-Matter development environment setup options
-********************************************
+1. Program the development kit for the Matter accessory device with one of available :ref:`matter_samples`.
+#. Configure the Thread Border Router on a Raspberry Pi.
+   See :ref:`ug_thread_tools_tbr` in the |NCS| documentation for details.
+#. Configure the mobile Matter controller:
 
-You can run the Matter controller on the same device as the Thread Border Router or run the Matter controller and the Thread Border Router on separate devices.
+   a. Build the controller using one of the options described in :ref:`ug_matter_configuring_controller_mobile`.
```
Comment (LuDuda) [inline doc/nrf/ug_matter_configuring_env.rst:70]:
```suggestion
   a. Install the controller using one of the options described in :ref:`ug_matter_configuring_controller_mobile`.
```

**Review by LuDuda — APPROVED** [review]
Thanks, looks good. Few nits below.

File: doc/nrf/ug_matter_configuring.rst (hunk 2 above)
Comment (greg-fer) [reply doc/nrf/ug_matter_configuring.rst:9]: Rephrased.

File: doc/nrf/ug_matter_configuring_protocol.rst (hunk 1 above)
Comment (greg-fer) [reply doc/nrf/ug_matter_configuring_protocol.rst:75]: No changes made.

File: doc/nrf/ug_matter_configuring.rst (hunk 3 above)
Comment (greg-fer) [reply doc/nrf/ug_matter_configuring.rst:6]: Changed.

File: doc/nrf/ug_matter_configuring.rst (hunk 4 above)
Comment (greg-fer) [reply doc/nrf/ug_matter_configuring.rst:35]: Edited.

File: doc/nrf/ug_matter_configuring.rst (hunk 5 above)
Comment (greg-fer) [reply doc/nrf/ug_matter_configuring.rst:41]: Edited

**Comment by greg-fer** [comment]
@Damian-Nordic , feel free to re-review.

File: doc/nrf/ug_matter_configuring_protocol.rst (hunk 1 above)
Comment (Damian-Nordic) [reply doc/nrf/ug_matter_configuring_protocol.rst:75]: Ok, I'll remember that.

**Review by Damian-Nordic — APPROVED** [review]

**Review by OeySa — APPROVED** [review]
This looks good to me!

**Pull request merged**